devops/scripts/benchmarks/utils/unitrace.py

```diff
@@ -0,0 +1,182 @@
+# Copyright (C) 2024-2025 Intel Corporation
+# Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+# See LICENSE.TXT
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+import os
+import shutil
+import re
+
+from options import options
+from utils.utils import run, git_clone
+from utils.oneapi import get_oneapi
+
+
+def extract_save_name_and_timestamp(dirname):
+    """
+    Extracts (save_name, timestamp) from a directory name of the form {save_name}_{timestamp},
+    where timestamp is always 15 characters: YYYYMMDD_HHMMSS.
+    save_name may contain underscores.
+    """
+    m = re.match(r"(.+)_(\d{8}_\d{6})$", dirname)
+    if m:
+        return m.group(1), m.group(2)
+    return None, None
+
+
+def prune_unitrace_dirs(base_dir, FILECNT=10):
+    """
+    Keeps only FILECNT newest directories for each save_name group in base_dir.
+    """
+    dirs = [d for d in os.listdir(base_dir) if os.path.isdir(os.path.join(base_dir, d))]
+    groups = {}
+    for d in dirs:
+        save_name, ts = extract_save_name_and_timestamp(d)
+        if save_name and ts:
+            groups.setdefault(save_name, []).append((d, ts))
+    for save_name, dirlist in groups.items():
+        # Sort by timestamp string (lexicographically, works for YYYYMMDD_HHMMSS)
+        dirlist.sort(key=lambda x: x[1])
+        if len(dirlist) > FILECNT:
+            for d, ts in dirlist[: len(dirlist) - FILECNT]:
+                full_path = os.path.join(base_dir, d)
+                print(f"Removing old unitrace dir: {full_path}")
+                shutil.rmtree(full_path)
+
+
+def unitrace_cleanup(bench_cwd, unitrace_output):
+    # Remove .pid files from the benchmark directory and .json files from cwd
+    unitrace_dir = os.path.dirname(unitrace_output)
+    unitrace_base = os.path.basename(unitrace_output)
+    print(f"Cleanup unitrace output {unitrace_base} from {unitrace_dir}")
+    for f in os.listdir(unitrace_dir):
+        if f.startswith(unitrace_base + "."):
+            os.remove(os.path.join(unitrace_dir, f))
+            print(f"Cleanup: Removed {f} from {unitrace_dir}")
+    if os.path.exists(bench_cwd):
+        for f in os.listdir(bench_cwd):
+            if f.endswith(".json"):
+                os.remove(os.path.join(bench_cwd, f))
+                print(f"Cleanup: Removed {f} from {bench_cwd}")
+
+
+def unitrace_prepare(name, unitrace_timestamp, command, extra_unitrace_opt=[]):
+    unitrace_bin = os.path.join(options.workdir, "unitrace-build", "unitrace")
+    if not os.path.exists(unitrace_bin):
+        raise FileNotFoundError(f"Unitrace binary not found: {unitrace_bin}. ")
+    os.makedirs(options.unitrace_res_dir, exist_ok=True)
+    if not options.save_name:
+        raise ValueError(
```

Comment:
I think this is raised too late. It should be raised before we set up all the benchmarks and try to run them. We already know whether unitrace and save_name has been set right after the arguments parsing.